source/frontend/src/features/mapSideBar/management/tabs/activities/edit/ManagementActivityEditContainer.tsx

```diff
@@ -61,19 +105,19 @@ export const ManagementActivityEditContainer: React.FunctionComponent<
   const onSave = async (model: ApiGen_Concepts_PropertyActivity) => {
     let result: ApiGen_Concepts_PropertyActivity | undefined = undefined;
     if (isValidId(model.id)) {
-      // TODO: implement update activity flow
+      result = await updateManagementActivity(managementFileId, model);
     } else {
       result = await addManagementActivity(managementFileId, model);
     }
 
-    if (exists(result)) {
+    if (exists(result) && isValidId(managementFileId)) {
       setStaleLastUpdatedBy(true);
       history.push(`/mapview/sidebar/management/${managementFileId}/activities/${result.id}`);
```

Comment:
Instead of hardcoding the path here, lets use the sidebarPathGenerator to create the routing path

Reply:
This is the first time I've heard of the "sidebarPathGenerator" - If this is something you created feel free to present it on our dev meeting so I can learn about it

Reply:
Also - we have a ton of "hardcoded" routes. If you created something more generic we should address all paths in a tech-debt task, not just this PR

Reply:
we talked about this some time ago. Without it, maintaining the routes is quite work-intensive.